ledger/src/blockstore.rs

```diff
@@ -1174,30 +1194,32 @@ impl Blockstore {
             .expect("Couldn't fetch from SlotMeta column family")
         {
             // Clear all slot related information
-            self.run_purge(slot, slot, PurgeType::Exact)
-                .expect("Purge database operations failed");
+            let run_purge_result = self.run_purge(slot, slot, PurgeType::Exact);
```

Comment:
> Also clearing the parent's next slots will leave this in a worst state if the shreds aren't also cleaned up, because repair won't happen

We could update `run_purge` to take a flag, and perform this cleanup of next_slots too. The benefit there would be that we could tie into the same `WriteBatch` to make the operation truly atomic (all succeeds or all fails, based on the write batch commit success)

Reply:
I'm not sure if `ReplayStage` should try to recover from any of the current panics here. It seems that if we have a rocksdb error here we should panic as blockstore is in a bad state.
I think the motivation is that if the rocksdb error happens to be during the `parent_slot_meta` cleanup, we should still reinsert the `child`'s `slot_meta` as upon restart we might be able to squeak by.

in retrospect it seems brittle to rely on this behavior so perhaps the change should be to only relax the parent's `slot_meta` cleanup and still fail fast on any rocksdb error.

Reply:
I like the idea of extending the use of `WriteBatch` to cover this whole function, it seems brittle to rely on specific orderings and recoveries, and better to say this function should be atomic.